### DISCUSSION TITLE
Impossible to make the Visual Studio PropertyGrid use my custom FileNameEditor derived class

### DISCUSSION THREAD

**Discussion by cadilhac** [dotnet/winforms#6581] [Q&A]
* .NET Core Version: 
.Net 6

* Have you experienced this same bug with .NET Framework?: 
No, it works perfectly with the .Net Framework

**Problem description:**
I want to use a custom FileNameEditor for a custom control property. I derived from FileNameEditor like so:

```
public class MyFileNameEditor : System.Windows.Forms.Design.FileNameEditor
{
    protected override void InitializeDialog(OpenFileDialog openFileDialog)
    {
        base.InitializeDialog(openFileDialog);

        openFileDialog.Filter = "My files (*.custom)|*.custom";
    }
}
```

I applied it like so:

```
[Editor(typeof(MyFileNameEditor ), typeof(System.Drawing.Design.UITypeEditor))]
public string MyFileName
{
    set [...]
    get [...]
}
```

In the VS designer, when selecting the custom control, inside the PropertyGrid, the property will have no editor.
Note that if I use the stock FileNameEditor instead, it just works.

**Comment by RussKie** [comment]
Unfortunately in .NET this is no longer possible - the designer is run as .NET process, and editors are run in the context of VS (net472 process). You can find more information in our latest blog post ["State of the Windows Forms Designer for .NET Applications"](https://devblogs.microsoft.com/dotnet/state-of-the-windows-forms-designer-for-net-applications/).
You'll need to modify your code and use [the designer extensibility SDK](https://www.nuget.org/packages/Microsoft.WinForms.Designer.SDK). We will be publishing more information in the near future, including how to author custom controls for the OOP designer.

**Reply by cadilhac** [reply]
> If the control brings its own Type Editor, then the required adjustments are more considerable. This is the same process the team underwent with the library of the standard controls: While the modal dialogs of a control’s designer can only work in the context of the Visual Studio process, the rest of the control’s designer runs in the context of the DesignToolServer’s process. That means a control with a custom type editor, which is shown in a modal dialog, always needs a Client/Server Control Designer combination. It needs to communicate between the modal UI in the Visual Studio process and the actual instance of the control in the DesignToolsServer process.

Thank you. Awaiting your documentation for adding custom UITypeEditors.
